Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx

```diff
@@ -106,17 +106,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
     useEffect(
         () => {
             const { allowedSubmissionTypes } = currentProblem || {};
-            const submissionType = first(allowedSubmissionTypes);
+            const submissionType = allowedSubmissionTypes?.find((st) => st.id === selectedSubmissionType?.id);
 
-            if (submissionType) {
-                const { id } = submissionType;
+            if (!submissionType) {
+                const defaultSubmissionType = first(allowedSubmissionTypes);
 
-                selectSubmissionTypeById(id);
-            } else {
-                selectSubmissionTypeById(null);
+                return isNil(defaultSubmissionType)
+                    ? selectSubmissionTypeById(null)
+                    : selectSubmissionTypeById(defaultSubmissionType.id);
```

Comment:
destruct defaultSubmissionType.id

Reply:
When I try to destructure it before the check if defaultSubmissionType is nil, it says the id does not exist on type ISubmissionTypeType | undefined. So I need to check first if the default submission type is undefined to destructure it afterwards, but I don't see a way to do it without involving a nested if.